crypto/threads_pthread.c

```diff
@@ -654,7 +654,19 @@ struct stack_traces {
 /* The glibc gettid() definition presents only since 2.30. */
 static ossl_inline pid_t get_tid(void)
 {
+#  ifdef OPENSSL_SYS_MACOSX
+    /*
+     * MACOS has the gettid call, but it does something completely different
+     * here than on other unixes.  Specifically it returns the uid of the calling thread
+     * (if set), or -1.  We need to use a MACOS specific call to get the thread id here
+     */
+    uint64_t tid;
+
+    pthread_threadid_np(NULL, &tid);
```

Comment:
A couple of questions,
Does this need a minimum version check, It looks like this didnt exist before 10.6?
I also cant see any docs on the range of the value returned in tid, it looks like pid_t is a signed 32 bit value?

Reply:
It does not need any minimum check because `get_tid()` is only used when `REPORT_RWLOCK_CONTENTION` is enabled. It's not perfect, but it's ok for our internal use.